Doc/library/typing.rst

```diff
@@ -623,27 +623,30 @@ These can be used as types in annotations using ``[]``, each having a unique syn
 
 .. data:: Union
 
-   Union type; ``Union[X, Y]`` is equivalent to ``X | Y`` and means either X or Y.
+   Union type; ``X | Y`` is equivalent to ``Union[X, Y]`` and means either X or Y.
 
-   To define a union, use e.g. ``Union[int, str]`` or the shorthand ``int | str``.  Details:
+   To define a union, use e.g. ``int | str`` or the older version ``Union[int, str]``.
```

Comment:
is it worth mentioning that `Union` and `|` behave a bit differently when used at runtime?

```
>>> type(Union[str, int])
<class 'typing._UnionGenericAlias'>
>>> type(str | int)
<class 'types.UnionType'>
```

Reply:
I get add a link to https://docs.python.org/3/library/stdtypes.html#union-type

Reply:
yes, I think that works ☺️

Reply:
Reading this, I would assume _equivalent to_ means that `X | Y == Union[X, Y]`. I recommend rephrasing to `logically equivalent` to resolve this.

Reply:
While they're technically equivalent (and will evaluate as equal), they're not identical. Would it be worth hinting at this by saying they're _logically_ equivalent?